src/main/kotlin/ai/blindspot/ktoolz/extensions/OtherExtensions.kt

```diff
@@ -87,3 +87,70 @@ inline fun <T : Any> T.applyIf(shouldApply: Boolean, block: T.() -> Unit): T {
+    if (this.length <= maxLength) return this
+    return this.substring(0, maxLength - ellipsis.length) + ellipsis
+}
+
+/**
+ * Creates a string like "className(details)", for example "Double(42.0)"
+ * Useful for implementing .toString() method (but in such case the first parameter MUST be passed).
+ */
+fun Any.toLongDebugString(description: String? = null, brackets: String = "()", className: String? = null): String {
+    val actualDescription = description ?: this.toString()
+    val actualClassName = className ?: this.javaClass.simpleName
+    val actualBrackets = if (brackets.length == 2) brackets else "<>"
+    return "$actualClassName${actualBrackets[0]}$actualDescription${actualBrackets[1]}"
+}
+
+/**
+ * Extracts the essential information from an object (most usually a string), whose toString() call result
+ * has a similar format as the output of function [toLongDebugString] .
+ * If the format is not similar, it simply returns this.toString() .
+ *
+ * For example, returns "42.0" for string "Double(42.0)".
+ * Returns "John 42, Peter 31" for string "EmployeeList{John 42, Peter 31}"
+ *
+ * Typical usage: the toString() method of a large collection-like object should contain only a very short description
+ *  of each collection item. This is done, for example, in [itemsToDebugString] ().
+ */
+fun Any?.toShortDebugString(): String {
+    val longString = this.toString()
+    if (longString.isEmpty()) return "EMPTY STRING" // Should not happen
+    val startsWithLetter = longString[0].let { it in 'a'..'z' || it in 'A'..'Z' }
+    if (!startsWithLetter) return longString // The format does not resemble an output of [toLongDebugString]
+
+    val bracketPairs = setOf("()", "[]", "<>", "{}")
+    for (pair in bracketPairs) {
+        if (longString.last() == pair[1]) {
+            val after = longString.substringAfter(pair[0])
+            if (after.length < 2) return longString
+            return after.substring(0, after.length - 1) // Omit the last char
+        }
+    }
+    return longString // The format does not resemble an output of [toLongDebugString]
+}
+
+/**
+ * Formats a collection to a readable string like "3 colors: red, yellow, green".
+ */
+inline fun <T> Collection<T>.itemsToDebugString(
+    itemsType: String = "items", // or "employees" etc.
+    separator: String = ", ", // inserted between (string representations of) collection items
+    itemLength: Int = 30, // max length
+    totalLength: Int = 200,
+    itemToString: (T) -> String = { item -> item.toShortDebugString() }
+): String {
+    val sb = StringBuilder("${this.size} $itemsType")
+    var currentSeparator = ": "  // we want to return something like: "3 items: item1, item2, item3"
+    for (item in this) {
+        val short: String = if (item == null) "NULL" else itemToString(item).restrictLengthWithEllipsis(itemLength)
+        if (short.length + sb.length >= totalLength) return sb.apply { append("…") }.toString()
```

Comment:
Now imagine that `item + short` has length equal to `totalLength` - then you append `...` and return string which has length bigger than `totalLength` which can be a bit confusing.

I propose two following solutions:
```kotlin
/**
 * Formats a collection to a readable string like "3 colors: red, yellow, green".
 */
inline fun <T> Collection<T>.itemsToDebugString(
    itemsType: String = "items", // or "employees" etc.
    separator: String = ", ", // inserted between (string representations of) collection items
    itemLength: Int = 30, // max length
    totalLength: Int = 200,
    crossinline itemToString: (T) -> String = { item -> item.toShortDebugString() }
) =  ("${this.size} $itemsType: " + this.joinToString(separator) { item ->
        if (item == null) "NULL"
        else itemToString(item).restrictLengthWithEllipsis(itemLength)
    }).restrictLengthWithEllipsis(totalLength)
```
and using `this` in inlined lambda
```kotlin
/**
 * Formats a collection to a readable string like "3 colors: red, yellow, green".
 */
inline fun <T : Any?> Collection<T>.itemsToDebugString(
    itemsType: String = "items", // or "employees" etc.
    separator: String = ", ", // inserted between (string representations of) collection items
    itemLength: Int = 30, // max length
    totalLength: Int = 200,
    crossinline itemToString: T.() -> String = { toShortDebugString() }
) = ("${this.size} $itemsType: " + this.joinToString(separator) {
        it?.itemToString()?.restrictLengthWithEllipsis(itemLength) ?: "NULL"
    }).restrictLengthWithEllipsis(totalLength)

```

Reply:
@Petr-Eichler  mm?

Reply:
I meant totalLength as an approximate length and that is more than sufficient for most debug output (and .toString() overrides). At that time I did not know that the code would be moved KToolz (or that something like KToolz exists :-) )  

But I fully admit that KTools deserves a high standard of quality. I will fix it if you don't like this solution:
`    itemLength: Int = 30, // max length
    totalLength: Int = 200, // approx max length `

Reply:
Lukáš, your suggested code example is very illustrative for me (although I don't know what would happen if I ommit `crossinline`). For the case of time efficiency when this.size is very high, maybe I should keep my current solution and call `restrictLengthWithEllipsis()` before the final return?

And all four my new functions should be `inline`, right?

Reply:
For the inlining - not all functions here should be inlined, only those using lambdas. There is no performance gain inlining standard functions and it might significantly bloat the compiled code.

For the code snippet - @LukasForst's solution is obviously much nicer, but in many cases you will produce strings with several thousands of characters (for long lists) only to throw out almost all of them in the next step. Fortunately, `joinToString` has a lot of options and you can specify the limit parameter directly. So just pass there the appropriate length and everything will be nice and shiny :)

Reply:
Yeah, I wanted to do something like that - but the [documentation](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/join-to-string.html) says that the parameter `limit` restricts the number of items joined to string, not the number of characters produced. 

So I think I should 
a) either keep my version and just `restrictLengthWithEllipsis` the result 
b) or to supply a "heuristical compromise" to the parameter `limit` (say `limit = 10` or `limit = totalLength / 5`). 
c) or use an absolutely reliable [limit](url), something like `1 + totalLength / (1 + separator.length)`
d) Or is there a better way?

Reply:
Nice catch, I did not read the documentation properly. I think it is ok to use the current implementation, just ensure that the total length will be always ensured (update the if statement). Using the `joinToString` method would produce unnecessarily long strings which we would have to shorten after that.